cmd/ctl/pkg/install/install.go

```diff
@@ -0,0 +1,230 @@
+)
+
+type InstallOptions struct {
+	settings  *cli.EnvSettings
+	client    *action.Install
+	cfg       *action.Configuration
+	valueOpts *values.Options
+
+	ChartName string
+	DryRun    bool
+
+	genericclioptions.IOStreams
+}
+
+const installCRDsFlagName = "installCRDs"
+const installDesc = `
+This command installs cert-manager. It uses the Helm libraries to do so.
+
+The latest published cert-manager chart on the "https://charts.jetstack.io" repo is used.
+Most of the features supported by 'helm install' are also supported by this command.
+In addition his command will always install CRD resources.
+
+Some example uses:
+	$ kubectl cert-manager x install
+or
+	$ kubectl cert-manager x install -n new-cert-manager
+or
+	$ kubectl cert-manager x install --version v1.4.0
+or
+	$ kubectl cert-manager x install --set prometheus.enabled=false
+
+To override values in the cert-manager chart, use either the '--values' flag and pass in a file
+or use the '--set' flag and pass configuration from the command line, to force
+a string value use '--set-string'. In case a value is large and therefore
+you want not to use neither '--values' nor '--set', use '--set-file' to read the
+single large value from file.
+`
+
+func NewCmdInstall(ctx context.Context, ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	settings := cli.New()
+	cfg := new(action.Configuration)
+
+	options := &InstallOptions{
+		settings:  settings,
+		cfg:       cfg,
+		client:    action.NewInstall(cfg),
+		valueOpts: &values.Options{},
+
+		IOStreams: ioStreams,
+	}
+
+	// Set default namespace cli flag value
+	defaults := make(map[string]string)
+	defaults["namespace"] = "cert-manager"
+
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "Install cert-manager",
+		Long:  installDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := helm.CopyCliFlags(cmd.Root().PersistentFlags(), defaults, settings); err != nil {
+				return nil
+			}
+			options.client.Namespace = settings.Namespace()
+
+			rel, err := options.runInstall(ctx)
+			if err != nil {
+				return err
+			}
+			return writeRelease(ioStreams.Out, rel, options.DryRun)
+		},
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	addInstallUninstallFlags(cmd.Flags(), &options.client.Timeout, &options.client.Wait)
+	addInstallFlags(cmd.Flags(), options.client)
+	addValueOptionsFlags(cmd.Flags(), options.valueOpts)
+	addChartPathOptionsFlags(cmd.Flags(), &options.client.ChartPathOptions)
+
+	cmd.Flags().BoolVar(&options.client.CreateNamespace, "create-namespace", true, "Create the release namespace if not present")
+	cmd.Flags().StringVar(&options.ChartName, "chart-name", "cert-manager", "Name of the chart to install")
+	cmd.Flags().BoolVar(&options.DryRun, "dry-run", false, "Simulate install and output manifest")
+
+	return cmd
+}
+
+func (o *InstallOptions) runInstall(ctx context.Context) (*release.Release, error) {
```

Comment:
Please add a couple of paragraphs in above this function explaining the strategy.
Something like:

> The overall strategy is to install the CRDs first, and not as part of a Helm release,
> and then to install a Helm release without the CRDs.
> This is to ensure that CRDs are not removed by a subsequent `helm uninstall` or by a future `kubectl cert-manager uninstall`.
> We want the removal of CRDs to only be performed by an administrator who understands that the consequences of removing CRDs will be the garbage collection of all the related CRs in the cluster.
> We first do a dry-run install of the chart (effectively `helm install --dry-run`) to both render the CRDs from the CRD templates in the Chart *and* to perform all the Helm pre-flight checks
> We then extract the CRDs from the resulting dry-run manifests and install those first. 
> Finally we perform a `helm install` to install the remaining non-CRD resources and wait for those to be "Ready".
> This creates a Helm "release" artifcact in a Secret in the target namespace, which contains a record of all the resources installed by Helm (except the CRDs).